### ISSUE TITLE
Bug in JointProbabilityDistribution.check_independence

### ISSUE THREAD

**Issue by khalibartan** [pgmpy/pgmpy#551]
In python 2 output is wrong while in python 3 output is correct .
Python 2:

```
In [1]: from pgmpy.factors import JointProbabilityDistribution as JPD

In [2]: import numpy as np

In [3]: prob = JPD(['x1','x2','x3'],[2,3,2], np.ones(12)/12)

In [4]: prob.check_independence('x1','x2')
Out[4]: False
```

Python 3

```
In [1]: from pgmpy.factors import JointProbabilityDistribution as JPD

In [2]: import numpy as np

In [3]: prob = JPD(['x1','x2','x3'],[2,3,2], np.ones(12)/12)

In [4]: prob.check_independence('x1','x2')
Out[4]: True
```

Couldn't quite figure out what caused this error because get_independencies is returning the right output(after the new fixes that i have done , just slight changes since it was not running ) and they both have similar code.

```
In [1]: from pgmpy.factors import JointProbabilityDistribution as JPD

In [2]: import numpy as np

In [3]: prob = JPD(['x1','x2','x3'],[2,3,2], np.ones(12)/12)

In [4]: prob.check_independence('x1','x2')
Out[4]: False

In [5]: prob.get_independencies()
Out[5]: 
(x1 _|_ x2)
(x1 _|_ x3)
(x2 _|_ x3)
```

Ping @ankurankan Take a look at the code, and tell me what can be the cause of the bug .

**Comment by khalibartan** [comment]
Found the bug, will be fixed in the new pr

**Comment by khalibartan** [comment]
@ankurankan When I was trying to dig in a little, I found that if condition was always true in **python2**, I found this by replacing `==` by `!=` and vice versa in both `check_independence` and `get_independence` and it returned the same result always.

**Comment by ankurankan** [comment]
@khalibartan So the issue here is that we are doing a `!=` comparison here: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/factors/JointProbabilityDistribution.py#L171. What I figured is that there is difference in the way python2 and 3 handle this. In case of python2 this statement wasn't even calling Factor's `__eq__` method. Whereas in case of python3 it was called. Here are the docs: https://docs.python.org/2/reference/datamodel.html https://docs.python.org/3/reference/datamodel.html for both python 2 and 3. 

A simple fix can be simply adding `__ne__` method in `Factor`:

``` python
    def __ne__(self, other):
        return not self.__eq__(other)
```

Just check if there is a better solution to this.

**Comment by khalibartan** [comment]
@ankurankan  Okay. I didn't knew this behaviour of python 2 . Will fix this in a new PR. I'll go with the same implementation (returning not of **eq**). I cannot think of any case where `==` and `!=` aren't of opposite truth value.

**Comment by khalibartan** [comment]
Fixed in #560 . Closed.
